src/store/store.ts

```diff
@@ -0,0 +1,1398 @@
+
+      return (currentShareClasses.length > 0)
+    },
+
+    /** The current filing name. */
+    getFilingName (): FilingNames {
+      if (this.isAlterationFiling) return FilingNames.ALTERATION
+      if (this.isCorrectionFiling) return FilingNames.CORRECTION
+      if (this.isFirmChangeFiling) return FilingNames.CHANGE_OF_REGISTRATION
+      if (this.isFirmConversionFiling) return FilingNames.CONVERSION
+      if (this.isRestorationFiling) {
+        if (this.isLimitedExtendRestorationFiling) return FilingNames.RESTORATION_EXTENSION
+        if (this.isLimitedConversionRestorationFiling) return FilingNames.RESTORATION_CONVERSION
+      }
+      if (this.isSpecialResolutionFiling) return FilingNames.SPECIAL_RESOLUTION
+      return null
+    },
+
+    /** The completing party data. */
+    getCompletingParty (): CompletingPartyIF {
+      return this.stateModel.completingParty
+    },
+
+    /** The special resolution object. */
+    getSpecialResolution (): SpecialResolutionIF {
+      return this.stateModel.specialResolution
+    },
+
+    /** The special resolution validity flags. */
+    getSpecialResolutionFormValid (): boolean {
+      return this.getValidationFlags.flagsCompanyInfo.isValidCreateSpecialResolution
+    },
+
+    /** The company info page validity flags. */
+    getSpecialResolutionConfirmValid (): boolean {
+      return this.getFlagsReviewCertify.isValidSpecialResolutionConfirm
+    },
+
+    /** The restoration object. */
+    getRestoration (): RestorationStateIF {
+      return this.stateModel.restoration
+    },
+
+    /** The restoration object. */
+    getStateFilingRestoration (): StateFilingRestorationIF {
+      return this.stateModel.stateFilingRestoration
+    },
+
+    /** The approval type validity. */
+    getApprovalTypeValid (): boolean {
+      return this.stateModel.validationFlags.flagsCompanyInfo.isValidApprovalType
+    },
+
+    /** The expiry date validity. */
+    getExpiryValid (): boolean {
+      return this.stateModel.validationFlags.flagsCompanyInfo.isValidExtensionTime
+    },
+
+    /** Returns false when users can change the sole proprietor (SP).
+     * Restricts the ability of non-staff users from changing the sole
+     * proprietor when the SP is an organization.  This restriction has been
+     * added to ensure that changes made in business-edit-ui are also updated by
+     * staff in COLIN */
+    hideChangeButtonForSoleProps (): boolean {
+      const isProprietor = this.getOrgPeople[0]?.roles[0]?.roleType === RoleTypes.PROPRIETOR
+      const isOrganization = this.getOrgPeople[0]?.officer?.partyType === PartyTypes.ORGANIZATION
+      const isDba = isProprietor && isOrganization
+      return !this.isRoleStaff && isDba
+    },
+
+    getExpiryDateString (): string {
+      return this.stateModel.restoration?.expiry
+    },
+
+    getFormattedExpiryText: (state) => (today = new Date()): string => {
```

Comment:
This was a tad annoying to deal with

Reply:
@seeker25 apologies for causing this annoyance.  Help me understand what you found annoying so I don't cause problems in the future.

Reply:
Tricky syntax - wasn't too bad though